File: app/src/types/board.ts
import type { Task } from './task'

export interface Board {
  id: string
  name: string
  teamId: string
  createdAt: string
}

export interface Column {
  id: string
  boardId: string
  name: string
  position: number
  tasks: Task[]
  createdAt: string
}

export interface BoardWithColumns extends Board {
  columns: Column[]
}

File: cmd/models/board.go
package models

import "time"

type Board struct {
	ID        string    `json:"id"`
	Name      string    `json:"name"`
	TeamID    string    `json:"teamId"`
	CreatedAt time.Time `json:"createdAt"`
}

type BoardWithColumns struct {
	Board
	Columns []Column `json:"columns"`
}
